### PR TITLE
Allow custom database loader

### PR DESCRIPTION
**Pull request by nemski**
#### Context

The database.yml file is an ERB, and hence we can evaluate data within it. For the scout plugin to do the evaluation it must have the necessary classes loaded.

#### Changes

Allow a custom loader to be specified that executes arbitrary code by calling the `load!` class method. For this to work the class name must be the camilised version of the file name.

#### Considerations

Does this defeat scout plugin code signing? I don't think so since the code is not downloaded from scout, but must be available within the app path

#### Tests

The tests were in a sad and sorry state. They failed from the beginning and once I did get them working I found a bug! `build_report` was mutating a constant, causing successive runs to fail.

gif

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -22,3 +22,7 @@ gem 'resque'
 gem 'nokogiri'
 gem 'memcache-client'
 gem 'sidekiq'
+gem 'rspec-core'
```
Comment (denmat) [inline Gemfile]: should these be in test group? they will probably fail to be found and cause errors in prod

File: delayed_job/monitor_delayed_jobs.rb [hunk 2]
```diff
@@ -97,13 +101,14 @@ def self.new_activerecord_version?
   }.freeze
 
   def build_report
+    custom_loader
```
Comment (denmat) [inline delayed_job/monitor_delayed_jobs.rb]: `custom_loader if custom_loader` or similar here?

**Comment by denmat** [comment]
Looks okay to me. :+1:  Thanks for doing this.

File: Gemfile (hunk 1 above)
Comment (nemski) [inline Gemfile]: Yep!

File: delayed_job/monitor_delayed_jobs.rb [hunk 3]
```diff
@@ -171,4 +193,9 @@ def queue_filter_sql
                             ''
                           end
   end
+
+  # Taken from http://infovore.org/archives/2006/08/11/writing-your-own-camelizer-in-ruby/
+  def camelize(str)
```
Comment (andrewjhumphrey) [inline delayed_job/monitor_delayed_jobs.rb]: It's sad that ruby doesn't have this natively.

**Comment by andrewjhumphrey** [comment]
+1 and a bonus +1 for the specs

**Pull request merged**